host/vm-manager/0001-Add-support-for-HPD.patch

```diff
@@ -0,0 +1,144 @@
+     { kGroupNet,     { kNetModel, kNetAdbPort, kNetFastbootPort } },
+     { kGroupVtpm,    { kVtpmBinPath, kVtpmDataDir } },
+     { kGroupRpmb,    { kRpmbBinPath, kRpmbDataDir } },
+diff --git a/src/guest/config_parser.h b/src/guest/config_parser.h
+index 947bc42..b793907 100644
+--- a/src/guest/config_parser.h
++++ b/src/guest/config_parser.h
+@@ -61,6 +61,9 @@ constexpr char kVgpuMonId[]    = "monitor";
+ constexpr char kVgpuOutputs[]    = "outputs";
+ 
+ constexpr char kDispOptions[] = "options";
++constexpr char kDispConnector0[] = "connectors_0";
++constexpr char kDispConnector1[] = "connectors_1";
++constexpr char kDispShowFps[] = "show_fps";
+ 
+ constexpr char kNetModel[] = "model";
+ constexpr char kNetAdbPort[] = "adb_port";
+diff --git a/src/guest/vm_builder_qemu.cc b/src/guest/vm_builder_qemu.cc
+index b06b51f..4f0bf3a 100644
+--- a/src/guest/vm_builder_qemu.cc
++++ b/src/guest/vm_builder_qemu.cc
+@@ -68,6 +68,17 @@ constexpr const unsigned int kExecQuantum = 25;
+ 
+ constexpr const char *kQmpPowerSocket = "/tmp/qmp-pwr-socket-";
+ 
++const std::string connectors[8] = {
++                "HDMI-1",
++                "HDMI-2",
++                "HDMI-3",
++                "HDMI-4",
++                "DP-1",
++                "DP-2",
++                "DP-3",
++                "DP-4",
++        };
++
+ static bool CheckUuid(std::string uuid) {
+     try {
+         boost::uuids::string_generator gen;
+@@ -281,6 +292,37 @@ bool VmBuilderQemu::SetupSriov(void) {
+         emul_cmd_.append(" -display gtk,gl=on,monitor=" + vgpu_mon_id);
+     }
+ 
++    int numberOfDisplays = 0;
++    std::string connectors_0 = cfg_.GetValue(kGroupDisplay, kDispConnector0);
++    std::string connectors_1 = cfg_.GetValue(kGroupDisplay, kDispConnector1);
++    std::string connector_str;
++    std::string show_fps = cfg_.GetValue(kGroupDisplay, kDispShowFps);
++    if (!show_fps.empty() && (show_fps.compare("on") == 0 || show_fps.compare("off") == 0)) {
++        emul_cmd_.append(",show-fps=" + show_fps + ",");
++    } else if (show_fps.empty()){
++        emul_cmd_.append(",show-fps=off,");
++    } else {
++	LOG(error) << "Please enter a valid value for show_fps.";
++	return false;
++    }
++
++    for(int i = 0; i < 8; i++) {
++        if(!connectors_0.empty() && connectors_0.compare(connectors[i]) == 0) {
++                        numberOfDisplays++;
++                        connector_str.append("connectors.0=");
++                        connector_str.append(connectors_0);
++                        connector_str.append(",");
++        }
++        if(!connectors_1.empty() && connectors_1.compare(connectors[i]) == 0) {
++                        numberOfDisplays++;
++                        connector_str.append("connectors.1=");
++                        connector_str.append(connectors_1);
++                        connector_str.append(",");
++        }
++    }
++
++    emul_cmd_.append(connector_str);
```

Comment:
This is the beginning of SetupSriov() method:

```
bool VmBuilderQemu::SetupSriov(void) {
    std::string vgpu_mon_id = cfg_.GetValue(kGroupVgpu, kVgpuMonId);
    if (vgpu_mon_id.empty()) {
        emul_cmd_.append(" -display gtk,gl=on");
    } else {
        emul_cmd_.append(" -display gtk,gl=on,monitor=" + vgpu_mon_id);

```

Can we do all the processing and error checking, then append display command to emul_cmd together in one logic instead of two separate places? This will be more coherent, more modular and easier to update in future.

Reply:
This is in vm-manager code. We should keep the changes here to minimum, as later we will move these changes to the their horizontal repo. Also, this is the process followed where they have appended to the same in multiple places.